lib/Color.h

```diff
@@ -49,6 +49,11 @@ class ColorRGBA
 		, a(ALPHA_OPAQUE)
 	{}
 
+    bool operator <(const ColorRGBA &val) const
+    {
+       return (r + g + b) < (val.r + val.g + val.b);
+    }
+
```

Comment:
Weird check. Perhaps instead provide this operator in place where it is needed? E.g. as functor to std::sort and such.

Reply:
Your current approach would break if r+g+b of different colors match.
RGB(255,0,0) and RGB(0,0,255) can't be added in the same std::map.

Checked code. 
I'd simply replace map<color, data> with vector of structs that contains color and data as fields.
More extensible and allows different plots with same color if needed.